packages/matrix-client-server/src/profiles/getProfiles.ts

```diff
@@ -111,38 +113,38 @@ export const getDisplayname = (
       typeof userId === 'string' &&
       userId.length > 0
     ) {
-      matrixDb
-        .get('profiles', ['displayname'], {
-          user_id: userId
-        })
-        .then((rows) => {
-          if (rows.length === 0) {
-            logger.info('No display_name found')
-            send(res, 404, errMsg('notFound', 'This user does not exist'))
-          } else {
-            if (rows[0].displayname === null) {
-              logger.info('No display_name found')
-              send(
-                res,
-                404,
-                errMsg('notFound', 'No display_name found for this user')
-              )
+      if (clientServer.isMine(userId)) {
+        clientServer.matrixDb
+          .get('profiles', ['displayname'], {
+            user_id: userId
+          })
+          .then((rows) => {
+            if (rows.length === 0) {
+              clientServer.logger.info('User not found')
+              send(res, 404, errMsg('notFound', 'User not found'))
             } else {
-              send(res, 200, {
-                displayname: rows[0].displayname
-              })
+              if (rows[0].displayname === null) {
+                send(res, 404, errMsg('notFound', 'Displayname not found'))
+              } else {
+                send(res, 200, {
+                  displayname: rows[0].displayname
+                })
+              }
             }
-          }
-        })
-        .catch((e) => {
-          /* istanbul ignore next */
-          logger.error('Error querying profiles:', e)
-          /* istanbul ignore next */
-          send(res, 500, errMsg('unknown', 'Error querying profiles'))
-        })
+          })
+          .catch((e) => {
+            /* istanbul ignore next */
+            send(res, 500, errMsg('unknown', e))
+            /* istanbul ignore next */
+            clientServer.logger.error('Error querying profiles:', e)
+          })
+      } else {
+        // TO DO : Have a look on remote server via federation
+        send(res, 500, errMsg('unknown', 'Cannot get profile of a remote user'))
```

Comment:
add return (security)

Reply:
when adding a return, I have a "unecessary return statement" eslint error
Do I still add it ?